### PR TITLE
Refactor: databases (front) part 1

### PR DESCRIPTION
**Pull request by fontanierh**
- Updates the core API bindings to match the new API
- Deletes most databases-related code (assistant builder, dust app blocks etc)
- Updates the CSV upload code (server & front) to work with the new API
- Resurrects CSV upload in labs

### CODE REVIEW

File: core/src/databases/database.rs [hunk 1]
```diff
@@ -200,7 +200,29 @@ impl Table {
         rows: &Vec<Row>,
         truncate: bool,
     ) -> Result<()> {
+        // Validate the tables schema, merge it if necessary and store it in the schema cache.
```
Comment (fontanierh) [inline core/src/databases/database.rs:203]: removed that by accident

File: front/components/sparkle/navigation.tsx [hunk 2]
```diff
@@ -316,6 +317,13 @@ export const subNavigationAdmin = ({
           href: `/w/${owner.sId}/u/extract`,
           current: current === "extract",
         },
+        {
+          id: "tables",
```
Comment (spolu) [inline front/components/sparkle/navigation.tsx:321]: As per IRL this is temporary as there is no reason for a Tables page at all anymore. Tables should be along documents in folders.

**Review by spolu — APPROVED** [review]

File: front/components/sparkle/navigation.tsx (hunk 2 above)
Comment (fontanierh) [reply front/components/sparkle/navigation.tsx:321]: Yes ✅

File: types/src/front/lib/core_api.ts [hunk 3]
```diff
@@ -77,33 +77,26 @@ type GetDatasetsResponse = {
...
   string,
   "int" | "float" | "text" | "bool" | "datetime"
 >;
 
-export type CoreAPIDatabaseTable = {
+export type CoreAPITable = {
   created: number;
-  database_id: string;
   table_id: string;
+  data_source_id: string;
```
Comment (spolu) [inline types/src/front/lib/core_api.ts:88]: Should be dataSourceName?

Comment (fontanierh) [reply types/src/front/lib/core_api.ts:88]: It's an object `returned by `core` so `dataSourceName` wouldn't make sense (that's a `front` thing)

**Pull request merged**